packages/dd-trace/src/profiling/exporters/agent.js

```diff
@@ -151,11 +151,17 @@ class AgentExporter {
           version
         },
         runtime: {
+          // os.availableParallelism only available in node 18.14.0/19.4.0 and above
+          available_processors: typeof os.availableParallelism === 'function'
```

Comment:
Note that if container has a cpu limit enforced through cpu CFS shares, the value will be wrong (eg. with a CPU limit of 0.5CPU on a 96 core system, value will be 96).

Reply:
Oh I see. This would still be useful, 'cause it could give an upper limit for a suggestion for `UV_THREADPOOL_SIZE`. The main use I have for this is that I could provide a profiler analysis: if there's more concurrent DNS and filesystem operations (and in future, cryptography and compression) than we have workers in the libuv worker thread pool, we can suggest the customer increase the size of the pool _as long_ as there's space for some extra parallelism in the CPU cores. K8s won't limit parallelism as looking from outside the box, you don't see the threads, it just limits the overall utilization of the CPU, right? Theory being that DNS and filesystem operations cause the libuv threads to be blocked on IO, so they're idle.

Reply:
But I could also definitely have use for `kubernetes.cpu.requests` and `kubernetes.cpu.limits` too. I don't suppose I can grab those values easily from within the Node app?

Reply:
I don't believe there is a way to get kubernetes.cpu.requests and kubernetes.cpu.limits from node app.
My point is that this value will be hard to interpret.
If threads are mostly blocked on network / IO operations, it might be beneficial to increase thread pool size even if available parallelism is low.